rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/window/NodeSourceWindow.java

```diff
@@ -385,6 +426,88 @@ protected void buildForm() {
+     */
+    protected void manageNodeSourceWindowItems(TextItem nodeSourceNameItem, CheckboxItem nodesRecoverableItem) {
+
+    }
+
+    /**
+     * Allow sub classes to redefine a custom behavior regarding what to do,
+     * regarding policies, in addition to the focused policy plugin.
+     */
+    protected void handleAdditionalPolicyFormItems(Map<String, String> selectItemValues,
+            PluginDescriptor focusedPolicyPlugin) {
+        Map<String, PluginDescriptor> allSupportedPolicies = this.controller.getModel().getSupportedPolicies();
+        addPluginValuesToAllFormItemOtherThanFocused(selectItemValues, focusedPolicyPlugin, allSupportedPolicies);
+    }
+
+    /**
+     * Allow sub classes to redefine a custom behavior regarding what to do,
+     * regarding infrastructures, in addition to the focused infrastructure
+     * plugin.
+     */
+    protected void handleAdditionalInfrastructureFormItems(Map<String, String> selectItemValues,
+            PluginDescriptor focusedInfrastructurePlugin) {
+        Map<String, PluginDescriptor> allSupportedInfrastructures = this.controller.getModel()
+                                                                                   .getSupportedInfrastructures();
+        addPluginValuesToAllFormItemOtherThanFocused(selectItemValues,
+                                                     focusedInfrastructurePlugin,
+                                                     allSupportedInfrastructures);
+    }
+
+    protected void fillFocusedPluginValues(LinkedHashMap<String, String> selectItemValues,
+            PluginDescriptor focusedPlugin) {
+        String pluginShortName = getPluginShortName(focusedPlugin);
+        selectItemValues.put(focusedPlugin.getPluginName(), pluginShortName);
+
+        List<FormItem> pluginFormItems = getPrefilledFormItems(focusedPlugin);
+        this.allFormItems.addAll(pluginFormItems);
+        this.allFormItemsPerPlugin.put(focusedPlugin.getPluginName(), pluginFormItems);
+    }
+
+    protected void addPluginValuesToAllFormItemOtherThanFocused(Map<String, String> selectItemValues,
+            PluginDescriptor focusedPlugin, Map<String, PluginDescriptor> allPluginDescriptors) {
+        for (Map.Entry<String, PluginDescriptor> entry : allPluginDescriptors.entrySet()) {
+            PluginDescriptor plugin = entry.getValue();
+            if (!plugin.getPluginName().equals(focusedPlugin.getPluginName())) {
+
+                selectItemValues.put(plugin.getPluginName(), getPluginShortName(plugin));
+                List<FormItem> currentPluginFormItems = getPrefilledFormItems(plugin);
+                this.allFormItems.addAll(currentPluginFormItems);
+                this.allFormItemsPerPlugin.put(plugin.getPluginName(), currentPluginFormItems);
+            }
+        }
+    }
+
     private void applyModificationsToNodeSource(VLayout nodeSourceWindowLayout, Label nodeSourcePluginsWaitingLabel,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)